### DISCUSSION TITLE
Can we guarantee ike sa initiation from one of the site as initiator only from our application ?

### DISCUSSION THREAD

**Discussion by hbhokare** [strongswan/strongswan#795] [Q&A]
Hi,
We are trying to develop the IPSec based application and calling different strongswan/davici API from our application. In our application, we restrict one of the site as an initiator only allowing ike sa init from that site based on some logic written in the application layer. We have the following questions related to ike-updown event getting generated with up= yes that is ike up event. 
Can you please answer following questions.
1. Can strongswan layer guarantee that we would receive ike up event with initiator=yes field on-site from which application is doing ike sa init conn ?
2. Are there any situations where we can see ike up event on site where we are not doing ike init conn from application layer ? 

Thanks in Advance !
-Harish

**Comment by tobiasbrunner** [comment]
>  In our application, we restrict one of the site as an initiator only allowing ike sa init from that site based on some logic written in the application layer.

You could just enable _charon.initiator_only_ in strongswan.conf, then any initiating packets are ignored completely (changing it currently does require a restart, though).

> Can strongswan layer guarantee that we would receive ike up event with initiator=yes field on-site from which application is doing ike sa init conn ?

Probably depends on how exactly initiating messages are dropped. With the above option, you'll never see an up event without `initiator=yes` as inbound IKE_SA_INIT requests are dropped as soon as the header of the packet has been parsed. And as initiator the flag will obviously be set for any established IKE_SA.

But note that the initiator flag can change during rekeying if it was initiated by the other end (that does not trigger an up event, but there could be an _ike-rekey_ event where the new IKE_SA has not set `initiator=yes` on the initiator of the replaced IKE_SA - and the eventual down event might also not have the flag set if such a rekeying happened).

> Are there any situations where we can see ike up event on site where we are not doing ike init conn from application layer ?

Not sure what you mean exactly, as you get the event for every established IKE_SA (i.e. as responder and initiator), just with different data/flags.
